Sources/HTTPManager.swift

```diff
@@ -1620,6 +1620,10 @@ extension HTTPManager {
             switch uploadBody {
             case .data(let data)?:
                 networkTask = inner.session.uploadTask(with: urlRequest, from: data)
+            case .formUrlEncoded(let queryItems)?:
+                networkTask = inner.session.uploadTask(with: urlRequest, from: FormURLEncoded.data(for: queryItems))
+            case .json(let json)?:
+                networkTask = inner.session.uploadTask(with: urlRequest, from: JSON.encodeAsData(json))
```

Comment:
And yeah, NSURLSession can't possibly put Content-Length in if you're doing streaming, because by definition it has no idea how long the stream is until it's actually streamed the whole thing.

The flag is awkward, but that's ok. The expectation is most servers don't need this (because HTTP 1.1 mandates support for Transfer-Encoding: chunked, so this only applies to HTTP 0.9 or HTTP 1.0 servers).